src/main/java/xbot/common/controls/actuators/wpi_adapters/CANTalonFxWpiAdapter.java

```diff
@@ -146,13 +164,14 @@ public void setPosition(Angle position) {
     }
 
     @Override
-    public void setPositionTarget(Angle position) {
-        setPositionTarget(position, 0);
-    }
-
-    @Override
-    public void setPositionTarget(Angle position, int slot) {
-        var controlRequest = new PositionDutyCycle(position).withSlot(slot);
+    public void setPositionTarget(Angle position, MotorPidMode mode, int slot) {
+        ControlRequest controlRequest;
+        switch (mode) {
+            case DutyCycle -> controlRequest = new PositionDutyCycle(position).withSlot(slot);
+            case Voltage -> controlRequest = new PositionVoltage(position).withSlot(slot);
+            case TrapezoidalVoltage -> controlRequest = new MotionMagicVoltage(position).withSlot(slot);
+            default -> throw new IllegalArgumentException("Unsupported mode: " + mode); // This should be impossible
```

Comment:
The only legal exception to throw in robot code is from the RobotAssertionManager - robot must always continue in degraded state, even in the face of a programming error.

Reply:
Repalced with alert